internal/hook/hook_tracker.go

```diff
@@ -48,37 +48,42 @@ type hookTrackerKey struct {
 	container string
 }
 
-// hookTrackerVal records the execution status of a specific hook.
-// hookTrackerVal is extensible to accommodate additional fields as needs develop.
-type hookTrackerVal struct {
+// hookStatus records the execution status of a specific hook.
+// hookStatus is extensible to accommodate additional fields as needs develop.
+type hookStatus struct {
 	// HookFailed indicates if hook failed to execute.
 	hookFailed bool
 	// hookExecuted indicates if hook already execute.
 	hookExecuted bool
+	// hookErr records error if hook execution fails
+	hookErr error
 }
 
 // HookTracker tracks all hooks' execution status
+// HookTracker is a map that uses the backup/restore name as the key
```

Comment:
Here the tracker is still being created per backup CR:
https://github.com/allenxu404/velero/blob/post_restore_hook_enhancement/pkg/backup/backup.go#L336
this should be fixed right?

Reply:
Yes.